### ISSUE TITLE
global not found in app still includes pages

### ISSUE THREAD

**Issue by darklight9811** [vercel/next.js#49796]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro Education
    Binaries:
      Node: 16.17.1
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.4.2
      eslint-config-next: 13.3.0
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.0.4
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

https://codesandbox.io/p/sandbox/zealous-scott-vppl6r

### To Reproduce

Create a global not found file inside app dir and build the project

### Describe the Bug

If you build an entire app only project, it still includes not found route using pages. (Even if I build a page 404, still includes pages dir).

### Expected Behavior

To only create files using the app dir (pages is also bringing a lot of js).

**Comment by JesseKoldewijn** [comment]
Hey man, 
you need to create a not-found.tsx file in your app folder instead of a page that has the /404 route based on the file structure.

https://nextjs.org/docs/app/api-reference/file-conventions/not-found

**Comment by darklight9811** [comment]
@JesseKoldewijn I also have a not found file, I made both to make sure it wouldn't "fix" the issue

**Comment by JesseKoldewijn** [comment]
> @JesseKoldewijn I also have a not found file, I made both to make sure it wouldn't "fix" the issue
From what I've noticed in my projects is that when using the not-found.tsx file it reports it during build as a page from the pages directory but actually build the file from the app dir. Have you checked your build/deploy results?

**Comment by JesseKoldewijn** [comment]
Just in codesandbox and in a couple of my own project this works as expected where it would just use the not-found.tsx file as the not found page.
![image](https://github.com/vercel/next.js/assets/60965908/cd49d53d-3b6d-44bb-85c9-62a7862ba3e7)

**Comment by darklight9811** [comment]
@JesseKoldewijn is not about running it, its about the built project
![image](https://github.com/vercel/next.js/assets/20667774/ea78562a-7c09-4902-aac4-e9dcaafab3b5)

**Comment by JesseKoldewijn** [comment]
> 

Yes I know, which is why I linked you that repo. If you check the deployment of it you'll see that it does use the not-found.tsx file but just reports it as a pages Route during the build display

**Comment by JesseKoldewijn** [comment]
@timneutkens Sorry for the @ mention. But do you know if there are any plans to fix this by chance? 😃

**Comment by darklight9811** [comment]
@JesseKoldewijn but are the files generated too? Even if they are not used.

**Comment by JesseKoldewijn** [comment]
> @JesseKoldewijn but are the files generated too? Even if they are not used.

What do you mean? it just generates a not found page based on the not-found.tsx file in the app dir as described. You can also see this if you check that repo I send before. It has the deployment on vercel and the url is located on the top right corner in the repo overview page

**Comment by timneutkens** [comment]
>  But do you know if there are any plans to fix this by chance

It'll be removed eventually, please be patient, there's much more important issues than an extra file being exported.

**Comment by JesseKoldewijn** [comment]
> > But do you know if there are any plans to fix this by chance
> 
> It'll be removed eventually, please be patient, there's much more important issues than an extra file being exported.

I agree, I do personally agree tho that it does look kind of confusing. But as long as it works, no issues in my eyes. 🎊

**Comment by icyJoseph** [comment]
Take for instance, this demo setup by a company, https://github.com/Makers-Den/next-bundle-test/issues/1, they thought that migrating to app dir doubled their bundles sizes... it was just the pages dir JS, plus the app dir JS that they were looking at, as a raw number, that lead them to that conclusion.
